auth0/provider.go

```diff
@@ -22,14 +22,24 @@ func Provider() *schema.Provider {
 				DefaultFunc: schema.EnvDefaultFunc("AUTH0_DOMAIN", nil),
 			},
 			"client_id": {
-				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("AUTH0_CLIENT_ID", nil),
+				Type:          schema.TypeString,
+				Optional:      true,
+				DefaultFunc:   schema.EnvDefaultFunc("AUTH0_CLIENT_ID", nil),
+				RequiredWith:  []string{"client_secret"},
+				ConflictsWith: []string{"api_token"},
 			},
 			"client_secret": {
-				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("AUTH0_CLIENT_SECRET", nil),
+				Type:          schema.TypeString,
+				Optional:      true,
+				DefaultFunc:   schema.EnvDefaultFunc("AUTH0_CLIENT_SECRET", nil),
+				RequiredWith:  []string{"client_id"},
+				ConflictsWith: []string{"api_token"},
+			},
+			"api_token": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				DefaultFunc:   schema.EnvDefaultFunc("api_token", nil),
```

Comment:
```suggestion
				DefaultFunc:   schema.EnvDefaultFunc("AUTH0_API_TOKEN", nil),
```

I think we by mistake renamed this incorrectly:)